File: Go/002/main.go
package main

import (
	"fmt"
	"time"
)

const (
	limit = 4_000_000
)

func solve(to int) int {
	answer := 0
	var n, m = 1, 1

	for m < to {
		m += n
		n = m - n
		if m%2 == 0 {
			answer += m
		}
	}

	return answer
}

func main() {
	start := time.Now()
	answer := solve(limit)
	elapsed := time.Since(start)

	fmt.Println(answer)
	fmt.Printf("%.6f seconds\n", elapsed.Seconds())
}

File: Python/002/main.py
import time


limit = 4_000_000


def solve(to):
    answer = 0
    x, y = 1, 1

    while y < to:
        y += x
        x = y - x

        if y % 2 == 0:
            answer += y

    return answer


start = time.time()
answer = solve(limit)
end = time.time()
print(answer)
print("--------------------------------------------------")
print(f"  {end - start:.6f} seconds")
